assets/schema_input.json

```
@@ -1,72 +1,74 @@
-        "errorMessage": "Tumor site should be number id or NA if sample is not tumor material"
-      },
-      "tumor_content": {
-        "type": "integer",
-        "errorMessage": "Tumor site should be number id or NA if sample is not tumor material",
-        "minimum": 0,
-        "maximum": 100
-      },
-      "run_id": {
-        "type": "string",
-        "pattern": "^\\S+$",
-        "errorMessage": "Run id must be provided and cannot contain spaces"
-      },
-      "path_to_run": {
-        "type": "string",
-        "errorMessage": "Path to run must be provided and exist",
-        "format": "directory-path",
-        "exists": true
-      },
-      "path_to_samplesheet": {
-        "type": "string",
-        "errorMessage": "Path to samplesheet",
-        "format": "file-path"
-      },
-      "barcode": {
-        "type": "string",
-        "errorMessage": "Run id must be provided and cannot contain spaces"
-      }
-    },
-    "required": [
-      "dataset_id",
-      "sample_id",
-      "molecule",
-      "sample_type",
-      "tumor_site",
-      "tumor_content",
-      "run_id",
-      "path_to_run"
-    ]
-  }
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "title": "inpred/tso500_nxf_workflow - params.input schema",
+    "description": "Schema for the file provided with params.input (ideally a samplesheet)",
+    "type": "array",
+    "items": {
+        "type": "object",
+        "properties": {
+            "dataset_id": {
+                "type": "string",
+                "pattern": "^\\S+$",
+                "errorMessage": "Dataset id must be provided and cannot contain spaces"
+            },
+            "sample_id": {
+                "type": "string",
+                "pattern": "^\\S+$",
+                "errorMessage": "Sample id must be provided and cannot contain spaces"
+            },
+            "molecule": {
+                "type": "string",
+                "pattern": "^[d,D,r,R]([n,N][a,A])?$",
+                "errorMessage": "Molecule is either D(NA) or R(NA), case insensitive"
+            },
+            "sample_type": {
+                "type": "string",
+                "pattern": "^(([t,T]([u,U][m,M][o,O][r,R])?)|([n,N]([o,O][r,R][m,M][a,A][l,L])?))$",
+                "errorMessage": "Sample type must be either T(umor) or N(ormal), case insensitive"
+            },
+            "tumor_site": {
+                "type": ["string", "integer"],
+                "pattern": "^(\\d{2}|XX)$",
+                "minimum": 10,
+                "maximum": 30,
+                "errorMessage": "Tumor site should be a two digit number id or XX if sample is not tumor material (refer to InPreD sample ID nomenclature v4 - https://tsoppi.readthedocs.io/en/latest/inpred_nomenclature.html)"
+            },
+            "tumor_content": {
+                "type": "integer",
+                "errorMessage": "Tumor site should be number id or NA if sample is not tumor material",
```

```suggestion
                "errorMessage": "Tumor content should be number id or NA if sample is not tumor material",
```

tumor content rather than tumor site?

---

done